### PR TITLE
feat: set new hours considering the offset

### PR DESCRIPTION
**Pull request by matheushahnn**
#### What does this PR do?
It fixes the issue related to timezone on Calendar

Closes https://github.com/grommet/grommet/issues/6421

#### How should this be manually tested?
Test this storybook https://storybook.grommet.io/?path=/story/input-dateinput-format-inline--format-inline

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/6421

### CODE REVIEW

**Comment by matheushahnn** [comment]
I still need to change on [grommet/src/js/components/Calendar/utils.js](https://github.com/grommet/grommet/blob/aad44bf01e4b1bff0ef8f34ddcefaf8e841b8921/src/js/components/Calendar/utils.js#L21)

**Comment by halocline** [comment]
@matheushahnn First, thank you for picking this issue up.

> I still need to change on [grommet/src/js/components/Calendar/utils.js](https://github.com/grommet/grommet/blob/aad44bf01e4b1bff0ef8f34ddcefaf8e841b8921/src/js/components/Calendar/utils.js#L21) In fact, I am not sure how it should do in this file, because it is adding days. Do you know where it is failing this utils and how I can reproduce locally?

This is actually performing as expected. I flagged this as an opportunity to consolidate logic as it looked like we are doing something somewhat related to the timezone offset work, but in a slightly different manner. I did not study closely.

If your judgement and recommendation is that this should remain as is, please let us know.

The rest of the work is looking great, thank you.

**Comment by matheushahnn** [comment]
I think we can keep it as it is.
Please, let me know what else I should do in this PR.

File: src/js/utils/dates.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const setHoursWithOffset = (date) => {
```
Comment (jcfilben) [inline src/js/utils/dates.js:1]: It looks like lines 89-95 of Calendar.js could be removed and we could just call this function instead

Comment (matheushahnn) [reply src/js/utils/dates.js:1]: Good call! 
Done

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**